packages/base/src/mainview/mainView.tsx

```diff
@@ -1101,16 +1152,50 @@ export class MainView extends React.Component<IProps, IStates> {
       .find(layer => layer.get('id') === id);
   }
 
+  /**
+   * Convenience method to get a specific layer index from OpenLayers Map
+   * @param id Layer to retrieve
+   */
+  private getLayerIndex(id: string) {
+    return this._Map
+      .getLayers()
+      .getArray()
+      .findIndex(layer => layer.get('id') === id);
+  }
+
   /**
    * Convenience method to get list layer IDs from the OpenLayers Map
    */
-  private getLayers() {
+  private getLayerIDs(): string[] {
     return this._Map
       .getLayers()
       .getArray()
       .map(layer => layer.get('id'));
   }
 
+  /**
+   * Move a layer in the stack.
+   *
+   * @param id - id of the layer.
+   * @param index - expected index of the layer.
+   */
+  moveLayer(id: string, index: number): void {
+    const currentIndex = this.getLayerIndex(id);
+    if (currentIndex === index || currentIndex === -1) {
+      return;
+    }
+    const layer = this.getLayer(id);
+    let nextIndex = index;
+    // should not be undefined since the id exists above
+    if (layer !== undefined) {
+      this._Map.getLayers().removeAt(currentIndex);
+      if (currentIndex < index) {
+        nextIndex -= 1;
+      }
+      this._Map.getLayers().insertAt(nextIndex, layer);
+    }
```

Comment:
```suggestion
    if (layer === undefined) {
    	return;
    }
    
    this._Map.getLayers().removeAt(currentIndex);
    if (currentIndex < index) {
      nextIndex -= 1;
    }
    this._Map.getLayers().insertAt(nextIndex, layer);
```

How do you feel about inverting this condition? I feel it makes the comment more clear and it's always nice to reduce indentation level :)